api/src/main/java/org/apache/iceberg/SnapshotUpdate.java

```diff
@@ -52,4 +53,13 @@
    */
   ThisT stageOnly();
 
+  /**
+   * Use a particular executor to plan. The default worker pool will be used by default.
+   *
+   * @param executorService the provided executor.
+   * @return this for method chaining.
+   */
+  default ThisT planWith(ExecutorService executorService) {
+    throw new UnsupportedOperationException();
```

Comment:
No need to throw `UnsupportedOperationException`. While this is an API, it is for user consumption. All of the implementations exist in core so we can just make sure our implementations are all up to date.